### PR TITLE
Add new audio and vision docs in Datasets blog post

### PR DESCRIPTION
**Pull request by stevhliu**
This blog post raises more awareness about the new audio and image sections in the Datasets docs which makes it easier for users to discover the necessary docs they need to work with audio and image datasets. Hopefully, this can also help shift the narrative from HF = NLP.

- [x] Add screenshot of new guide overview.

### CODE REVIEW

**Review by osanseviero — COMMENTED** [review]
Looks good to me!

File: datasets-docs-update.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+---
+title: "Introducing new audio and vision documentation in 🤗 Datasets"
+thumbnail: assets/87_datasets-docs-update/thumbnail.gif
```
[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets-docs-update.md:3]: omg love it !

File: datasets-docs-update.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    <a href="/stevhliu"> 
+        <img class="avatar avatar-user" src="https://aeiljuispo.cloudimg.io/v7/https://s3.amazonaws.com/moonup/production/uploads/1599079986463-noauth.jpeg?w=200&h=200&f=face" title="Gravatar">
+        <div class="bfc">
+            <code>stevhliu</code>
+            <span class="fullname">Steven Liu</span>
+        </div>
+    </a>
+</div>
+
+Open and reproducible datasets are essential for advancing good machine learning. At the same time, datasets have grown tremendously in size as rocket fuel for large language models. In 2020 we launched `nlp`, a library dedicated to:
```
[the remark this answers is not in the document]
Comment (lhoestq) [reply datasets-docs-update.md:27]: yea not sure it's that interesting to mention that it was named "nlp" at that time

[the remark this answers is not in the document]
Comment (julien-c) [reply datasets-docs-update.md]: indeed, let's please forget that phase 🤣  (@thomwolf 😡 )

File: datasets-docs-update.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+</div>
+</div>
+
+The 🤗 Datasets team has been building tools and features to make working with these dataset types as simple as possible for the best user experience. We’ve added some new documentation along the way to help you learn more about loading and processing audio and image datasets.
+
+## Quickstart
+
+The [Quickstart](https://huggingface.co/docs/datasets/master/en/quickstart) is one of the first places new users visit for a TLDR about a library’s features. That’s why we needed to update the Quickstart to include how you can use 🤗 Datasets to work with audio and image datasets. Choose a dataset modality you want to work with and see an end-to-end example of how to load and process the dataset to get it ready for training with either PyTorch or TensorFlow.
```
Comment (nateraw) [inline datasets-docs-update.md:47]:
```suggestion
The [Quickstart](https://huggingface.co/docs/datasets/quickstart) is one of the first places new users visit for a TLDR about a library’s features. That’s why we needed to update the Quickstart to include how you can use 🤗 Datasets to work with audio and image datasets. Choose a dataset modality you want to work with and see an end-to-end example of how to load and process the dataset to get it ready for training with either PyTorch or TensorFlow.
```

File: datasets-docs-update.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+Also new in our Quickstart is the `to_tf_dataset` function which takes care of converting a dataset into a `tf.data.Dataset` like a mama bear taking care of her cubs. This means you don’t have to write any code yourself - shuffle and load batches of data - to get your dataset to play nicely with TensorFlow. Once you’ve converted your dataset into a `tf.data.Dataset`, you can train your model with the usual TensorFlow or Keras methods.
```
Comment (nateraw) [inline datasets-docs-update.md:49]: > This means you don’t have to write any code yourself - shuffle and load batches of data - to get your dataset to play nicely with TensorFlow.

I like where this is going but I think it feels a little awkward as is. "shuffle and load batches of data" doesn't sound right grammatically to me. Maybe something like "like shuffling and loading batches of data" (that's not 100% there either)

File: datasets-docs-update.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+Check out the [Quickstart](https://huggingface.co/docs/datasets/master/en/quickstart) today to learn how to work with different dataset modalities and try out the new `to_tf_dataset` function!
```
Comment (nateraw) [inline datasets-docs-update.md:51]:
```suggestion
Check out the [Quickstart](https://huggingface.co/docs/datasets/quickstart) today to learn how to work with different dataset modalities and try out the new `to_tf_dataset` function!
```

File: datasets-docs-update.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+To make all of the modality-specific documentation more discoverable, we created dedicated sections with guides focused on showing you how to load and process each modality. If you’re looking for specific information about working with a dataset modality, take a look at these dedicated sections first. Meanwhile, functions that are non-specific and can be used broadly are documented in the General Usage section. Reorganizing the documentation in this way will allow us to better scale to other dataset types we plan to support in the future.
+
+<figure class="image table text-center m-0 w-full">
+  <img style="border:none;" alt="An overview of the how-to guides page that displays five new sections of the guides: general usage, audio, vision, text, and dataset repository." src="assets/87_datasets-docs-update/overview.png" />
+  <figcaption>The guides are organized into sections that cover the most essential aspects of 🤗 Datasets.</figcaption>
+</figure>
+
+Check out the [dedicated guides](https://huggingface.co/docs/datasets/master/en/how_to) to learn more about loading and processing datasets for different modalities.
```
Comment (nateraw) [inline datasets-docs-update.md:69]:
```suggestion
Check out the [dedicated guides](https://huggingface.co/docs/datasets/how_to) to learn more about loading and processing datasets for different modalities.
```

File: datasets-docs-update.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+## ImageFolder
+
+Typically, 🤗 Datasets users [write a dataset loading script](https://huggingface.co/docs/datasets/master/en/dataset_script) to download and generate a dataset with the appropriate `train` and `test` splits. With the `ImageFolder` dataset builder, you don’t need to write any code to download and generate an image dataset. Loading an image dataset for image classification is as simple as ensuring your dataset is organized in a folder like:
```
Comment (nateraw) [inline datasets-docs-update.md:73]:
```suggestion
Typically, 🤗 Datasets users [write a dataset loading script](https://huggingface.co/docs/datasets/dataset_script) to download and generate a dataset with the appropriate `train` and `test` splits. With the `ImageFolder` dataset builder, you don’t need to write any code to download and generate an image dataset. Loading an image dataset for image classification is as simple as ensuring your dataset is organized in a folder like:
```

File: datasets-docs-update.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+{"file_name": "0001.png", "objects": {"bbox": [[302.0, 109.0, 73.0, 52.0]], "categories": [0]}}
+{"file_name": "0002.png", "objects": {"bbox": [[810.0, 100.0, 57.0, 28.0]], "categories": [1]}}
+{"file_name": "0003.png", "objects": {"bbox": [[160.0, 31.0, 248.0, 616.0], [741.0, 68.0, 202.0, 401.0]], "categories": [2, 2]}}
+
+dataset = load_dataset("imagefolder", data_dir="/path/to/folder", split="train")
+dataset[0]["objects"]
+{"bbox": [[302.0, 109.0, 73.0, 52.0]], "categories": [0]}
+```
+
+You can use `ImageFolder` to load an image dataset for nearly any type of image task if you have a metadata file with the required information. Check out the [ImageFolder](https://huggingface.co/docs/datasets/master/en/image_process) guide to learn more.
```
Comment (nateraw) [inline datasets-docs-update.md:104]:
```suggestion
You can use `ImageFolder` to load an image dataset for nearly any type of image task if you have a metadata file with the required information. Check out the [ImageFolder](https://huggingface.co/docs/datasets/image_load) guide to learn more.
```

Note I'm linking to image_load not image_process in this one as I think that's the one you meant to put

File: datasets-docs-update.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+## What’s next?
+
+Similar to how the first iteration of our `datasets` library standardized text datasets and made them super easy to download and process, we are very excited to bring this same level of user-friendliness to audio and image datasets. In doing so, we hope it’ll be easier for users to train, build, and evaluate models and applications across all different modalities.
+
+In the coming months, we’ll continue to add new features and tools to support working with audio and image datasets. Word on the 🤗 Hugging Face street is that there’ll be something called `AudioFolder` coming soon! 🤫 While you wait, feel free to take a look at the [audio processing guide](https://huggingface.co/docs/datasets/master/en/audio_process) and then get hands-on with an audio dataset like [GigaSpeech](https://huggingface.co/datasets/speechcolab/gigaspeech).
```
Comment (nateraw) [inline datasets-docs-update.md:110]:
```suggestion
In the coming months, we’ll continue to add new features and tools to support working with audio and image datasets. Word on the 🤗 Hugging Face street is that there’ll be something called `AudioFolder` coming soon! 🤫 While you wait, feel free to take a look at the [audio processing guide](https://huggingface.co/docs/datasets/audio_process) and then get hands-on with an audio dataset like [GigaSpeech](https://huggingface.co/datasets/speechcolab/gigaspeech).
```

File: datasets-docs-update.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+---
+
+Join the [forum](https://discuss.huggingface.co/) for any questions and feedback about working with audio and image datasets. If you discover any bugs, please open a [GitHub Issue](https://github.com/huggingface/datasets/issues/new/choose), so we can take care of it.
+
+Feeling a little more adventurous? Contribute to the growing community-driven collection of audio and image datasets on the [Hub](https://huggingface.co/datasets)! [Create a dataset repository](https://huggingface.co/docs/datasets/master/en/upload_dataset) on the Hub and upload your dataset. If you need a hand, open a discussion on your repository’s **Community tab** and ping one of the 🤗 Datasets team members to help you cross the finish line!
```
Comment (nateraw) [inline datasets-docs-update.md:116]:
```suggestion
Feeling a little more adventurous? Contribute to the growing community-driven collection of audio and image datasets on the [Hub](https://huggingface.co/datasets)! [Create a dataset repository](https://huggingface.co/docs/datasets/upload_dataset) on the Hub and upload your dataset. If you need a hand, open a discussion on your repository’s **Community tab** and ping one of the 🤗 Datasets team members to help you cross the finish line!
```

**Review by nateraw — COMMENTED** [review]
Few more dead links updated here. Other than that, looks good to me :)

File: datasets-docs-update.md (hunk 4 above)
Comment (stevhliu) [reply datasets-docs-update.md:49]: Haha you're right. Instead of trying to be cute, I'll just get rid of the `-`.

File: _blog.yml [hunk 11]
```diff
@@ -977,4 +977,15 @@
...
     - nlp
-    - guide
\ No newline at end of file
+    - guide
+
+- local: datasets-docs-update
+  title: "Introducing new audio and vision documentation in 🤗 Datasets"
+  author: stevhliu
+  thumbnail: assets/87_datasets-docs-update/thumbnail.gif
+  date: July 11, 2022
```
Comment (osanseviero) [inline _blog.yml:986]: To update before merge

File: datasets-docs-update.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+    <a href="/stevhliu"> 
+        <img class="avatar avatar-user" src="https://aeiljuispo.cloudimg.io/v7/https://s3.amazonaws.com/moonup/production/uploads/1599079986463-noauth.jpeg?w=200&h=200&f=face" title="Gravatar">
+        <div class="bfc">
+            <code>stevhliu</code>
+            <span class="fullname">Steven Liu</span>
+        </div>
+    </a>
+</div>
+
+Open and reproducible datasets are essential for advancing good machine learning. At the same time, datasets have grown tremendously in size as rocket fuel for large language models. In 2020 we launched `datasets`, a library dedicated to:
```
Comment (osanseviero) [inline datasets-docs-update.md:27]:
```suggestion
Open and reproducible datasets are essential for advancing good machine learning. At the same time, datasets have grown tremendously in size as rocket fuel for large language models. In 2020, Hugging Face launched `datasets`, a library dedicated to:
```

File: datasets-docs-update.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+
+1. Providing access to standardized datasets with a single line of code.
+2. Tools for rapidly and efficiently processing large-scale datasets.
+
+Thanks to the community, we added hundreds of NLP datasets in many languages and dialects during the [Datasets Sprint](https://discuss.huggingface.co/t/open-to-the-community-one-week-team-effort-to-reach-v2-0-of-hf-datasets-library/2176)! 🤗 ❤️
+
+But text datasets are just the beginning. Data is represented in richer formats like 🎵 audio, 📸 image, and even a combination of audio and text or image and text. Models trained on these datasets enable awesome applications like creating a 3D semantic scene from a single monocular RGB image!
```
Comment (osanseviero) [inline datasets-docs-update.md:34]:
```suggestion
But text datasets are just the beginning. Data is represented in richer formats like 🎵 audio, 📸 images, and even a combination of audio and text or image and text. Models trained on these datasets enable awesome applications like creating a 3D semantic scene from a single monocular RGB image!
```

File: datasets-docs-update.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+
+<div class="hidden xl:block">
+<div style="display: flex; flex-direction: column; align-items: center;">
+<iframe src="https://hf.space/embed/CVPR/MonoScene/+
```
Comment (osanseviero) [inline datasets-docs-update.md:38]: This space unfortunately gives me an error, I would consider picking another that is fast :smile:

File: datasets-docs-update.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+" frameBorder="0" width="1400" height="690" title="Gradio app" class="p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+</div>
+</div>
+
+The 🤗 Datasets team has been building tools and features to make working with these dataset types as simple as possible for the best user experience. We’ve added some new documentation along the way to help you learn more about loading and processing audio and image datasets.
```
Comment (osanseviero) [inline datasets-docs-update.md:43]:
```suggestion
The 🤗 Datasets team has been building tools and features to make working with these dataset types as simple as possible for the best developer experience. There is new documentation along the way to help you learn more about loading and processing audio and image datasets.
```

File: datasets-docs-update.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+## Quickstart
+
+The [Quickstart](https://huggingface.co/docs/datasets/quickstart) is one of the first places new users visit for a TLDR about a library’s features. That’s why we needed to update the Quickstart to include how you can use 🤗 Datasets to work with audio and image datasets. Choose a dataset modality you want to work with and see an end-to-end example of how to load and process the dataset to get it ready for training with either PyTorch or TensorFlow.
```
Comment (osanseviero) [inline datasets-docs-update.md:47]:
```suggestion
The [Quickstart](https://huggingface.co/docs/datasets/quickstart) is one of the first places new users visit for a TLDR about a library’s features. That’s why we updated the Quickstart to include how you can use 🤗 Datasets to work with audio and image datasets. Choose a dataset modality you want to work with and see an end-to-end example of how to load and process the dataset to get it ready for training with either PyTorch or TensorFlow.
```

File: datasets-docs-update.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+
+Also new in our Quickstart is the `to_tf_dataset` function which takes care of converting a dataset into a `tf.data.Dataset` like a mama bear taking care of her cubs. This means you don’t have to write any code to shuffle and load batches from your dataset to get it to play nicely with TensorFlow. Once you’ve converted your dataset into a `tf.data.Dataset`, you can train your model with the usual TensorFlow or Keras methods.
```
Comment (osanseviero) [inline datasets-docs-update.md:49]:
```suggestion
Also new in the Quickstart is the `to_tf_dataset` function which takes care of converting a dataset into a `tf.data.Dataset` like a mama bear taking care of her cubs. This means you don’t have to write any code to shuffle and load batches from your dataset to get it to play nicely with TensorFlow. Once you’ve converted your dataset into a `tf.data.Dataset`, you can train your model with the usual TensorFlow or Keras methods.
```

File: datasets-docs-update.md (hunk 14 above)
Comment (osanseviero) [reply datasets-docs-update.md:38]: It worked after a factory reboot though :thinking:

File: datasets-docs-update.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+<figure class="image table text-center m-0 w-full">
+  <img style="border:none;" alt="Cards with links to end-to-end examples for how to process audio, vision, and NLP datasets" src="assets/87_datasets-docs-update/quickstart.png" />
+  <figcaption>Choose your dataset adventure!</figcaption>
+</figure>
+
+## Dedicated guides
+
+Each dataset modality has specific nuances on how to load and process them. For example, when you load an audio dataset, the audio signal is automatically decoded and resampled on-the-fly by the `Audio` feature. This is quite different from loading a text dataset!
+
+To make all of the modality-specific documentation more discoverable, we created dedicated sections with guides focused on showing you how to load and process each modality. If you’re looking for specific information about working with a dataset modality, take a look at these dedicated sections first. Meanwhile, functions that are non-specific and can be used broadly are documented in the General Usage section. Reorganizing the documentation in this way will allow us to better scale to other dataset types we plan to support in the future.
```
Comment (osanseviero) [inline datasets-docs-update.md:62]:
```suggestion
To make all of the modality-specific documentation more discoverable, there are new dedicated sections with guides focused on showing you how to load and process each modality. If you’re looking for specific information about working with a dataset modality, take a look at these dedicated sections first. Meanwhile, functions that are non-specific and can be used broadly are documented in the General Usage section. Reorganizing the documentation in this way will allow us to better scale to other dataset types we plan to support in the future.
```

File: datasets-docs-update.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+<figure class="image table text-center m-0 w-full">
+  <img style="border:none;" alt="A table of images of dogs and their associated label." src="assets/87_datasets-docs-update/good_boi_pics.png" />
+  <figcaption>Your 🐶 dataset should look something like this once you've uploaded it to the Hub and preview it.</figcaption>
+</figure>
+
+Image labels are generated in a `label` column based on the directory name. `ImageFolder` allows you to get started instantly with an image dataset, eliminating the time and effort required to write a dataset loading script.
+
+But wait, it gets even better. If you have a file containing some metadata about your image dataset, `ImageFolder` can be used for other image tasks like image captioning and object detection. For example, object detection datasets commonly have *bounding boxes*, coordinates in an image that identify where an object is. `ImageFolder` can use this file to link the metadata about the bounding box and category for each image to the corresponding images in the folder:
```
Comment (osanseviero) [inline datasets-docs-update.md:92]:
```suggestion
But wait, it gets even better! If you have a file containing some metadata about your image dataset, `ImageFolder` can be used for other image tasks like image captioning and object detection. For example, object detection datasets commonly have *bounding boxes*, coordinates in an image that identify where an object is. `ImageFolder` can use this file to link the metadata about the bounding box and category for each image to the corresponding images in the folder:
```

**Review by osanseviero — APPROVED** [review]
Cool blog post! I left some nits but looks great!

One thing I saw is that sometimes you say :hugs: Datasets and sometimes :hugs: `datasets` and sometimes just `datasets`, so I would try to be consistent. 

Thanks for the blog post!

**Pull request merged**